spec/lib/inventory_dashboard_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+require "inventory_dashboard.rb"
+
+RSpec.describe InventoryDashboard do
+  let!(:item) { FactoryGirl.create(:item) }
+  let!(:category) { FactoryGirl.create(:category) }
+
+  subject { InventoryDashboard.new(Item.all) }
+
+  describe "#initialize" do
+    it "is initialized with all valid items" do
+      expect(subject.items).to include item
+    end
+  end
+
+  describe "#new_item" do
+    it "creates a new item" do
+      expect(subject.new_item(name: "N", shelf_life_days: 2, quantity: 1, price: 2)).to respond_to(:save)
```

Comment:
You probably also want to make sure it actually creates what you need, with `Item.count` Something like:

````ruby
expect { subject.new_item(name: "N", shelf_life_days: 2, quantity: 1, price: 2) }.to change { Item.count }.by(1)
````